src/LearnositySdk/Utils/Json.php

```diff
@@ -104,4 +111,67 @@ public static function isJson($val)
     {
         return !is_null(json_decode($val, true));
     }
+
+    /**
+     * This function builds up a map of float values in scientific notation and the equivalent float
+     * notations. json_encode encodes float values in scientific notation, but we want to make sure
+     * they are in float notation
+     *
+     * @param $json the $json to be encoded
+     * @param $floatMap pointer to the floatMap
+     */
+    private static function getHighPrecisionFloatMap($json, &$floatMap)
+    {
+        if (is_array($json)) {
+            foreach ($json as $value) {
+                if (is_array($value)) {
+                    self::getHighPrecisionFloatMap($value, $floatMap);
+                }
+                if (is_float($value)) {
```

Comment:
Actually, you can get rid of all of that, and just have something like

```
foreach ($json as $value) {
   $currentMap[] = static::getHighPrecisionFloatMap($value, $floatMap);
   if(!empty($currentMap)) {
     floatMap[] = $currentMap;
   }
}
```

(This code assumes all my other suggestions, I haven't checked it, but you should get the idea.)

Reply:
@shtrom I added this because our unit tests try to encode non json values: https://github.com/Learnosity/learnosity-sdk-php/blob/master/src/tests/LearnositySdk/Utils/JsonTest.php#L18 Therefore I added the 'is_float' in the foreach and then again outside of it. I'm assuming that the test cases test a valid scenario, so I want to make sure it's handled properly. Alternatively, we could just change the unit tests and only call the function with valid json.